### ISSUE TITLE
ReStock vs Ven's in 1.6 and Beyond

### ISSUE THREAD

**Issue by pap1723** [KSP-RO/RP-0#1078]
I am leaning towards making Restock a dependency for the 1.6+ versions of RP-1. We currently rely on Ven's to replace the "not great stock items" but it is not a mod that has seen a real release on CKAN since the 1.3.1 days. It still has some updates and can be a manual download, but the quality of the stuff is not as good as Restock. I went through all of the parts for Ven's vs Restock and have my notes below.

Here is my proposal...(but I can talked out of this if you have compelling reasons why)
* Restock becomes an RP-1 dependency
  * The quality of the work is excellent, and one thing they have done well is to make variants instead of outright replacements on some of the parts. So there are some texture variants for the probe parts that are a nice bonus.
* Ven's becomes "integrated" into RP-1
  * For the parts that are not included anywhere else, we will adopt them into RP-1 and distribute them along with it. Ven's license allows it as they released all of their stuff when they left the community.
  * If we determine that the Ven's options are better than Restock, we can add them as a variant on the parts.

I have gone through all of the Ven's, Restock and Stock parts and have comments on them below. Please let me know your thoughts. All of the ones I have listed below as **integrate** come out to 31 parts. This adds a lot of overhead to RP-1 and is not ideal. I guess we could package it as a separate RPO-Vens mod?

The challenge of trying to support both Restock and Ven's is how the model replacements are handled by both mods. We could do it, and it might be the best option, but it will take a lot of wrangling and changes to get it to work.

### Ven's Additional Parts
These are the parts that do not exist in stock, so we need to decide whether to integrate them into RP-1 or not. I have my recommendations on them, but will listen to all ideas.

**CREW**
* VA Command Pod (I defer to popular opinion, do people use it?)
* Kerb Can (**integrate**, nice for stations)
* Large Inflatable HAB (**integrate**, nice for stations)

**ENGINES**
* Rattler Jet Engine (I defer to popular opinion)
* Fire Ant Engine (**integrate** into RP-1 as this is the model for so many things)
* Dachshund Engine (**integrate**, it is the RD100)
* Shiba Engine (remove, not currently used)
* Huff MP Engine (remove, not currently used)
* Breeze MP Engine (**integrate**, it is the Astris)
* Bollard Engine (I defer to popular opinion, it is the RD108, but SXT has one as well)
* Pila Engine (**integrate**, it is the LR91)
* Nova Atomic Engine (remove, not currently used)
* Thunder Engine (I defer to popular opinion, it is the BE4 currently)
* RockoMAX SRB (**integrate**, it is the UA1205)
* SnubOtron (**integrate**, it is a Sep Motor)
* Cricket SRB (**integrate**, it is a Castor-4)
* Grasshopper SRB (**integrate**, it is a Castor-2)
* Sickle SRB (remove, not currently used)

**CONTROL**
* Radial Avionics (I defer to popular opinion, it is currently a radial mounted avionics unit, but I've never used it)
* Inline RCS Block (I defer to popular opinion)
* Radial Reaction Wheel (**integrate**, nice for probes)
* 45 Degree RCS (**integrate**, at least until Restock Plus Support is added)
* Heavy RCS (**integrate**, at least until Restock Plus Support is added)
* Adv. Reaction Wheel (remove, not currently used)

**STRUCTURAL**
* 16 parts, **integrate** all

**OTHER**
* Parachute Docking Port (remove, not currently used)
* Small Radial Decoupler (**integrate**, nice to have)
* Mk1 Cargo Bay (I defer to popular opinion, I have never used it)
* OCTO Service Bay (I defer to popular opinion, I have never used it)
* Large Landing Leg (remove, not currently used)

**POWER AND COMMS**
* Solar Panels (**integrate**, nice to have)
* Mini RTG (**integrate**, nice to have)
* Communotron 88-88-V (remove, not currently used -> Pending Comms Overhaul?)

**UTILITY**
* Mk3 Spotlight, Mk4 Light, Mk5 Strip Light (I defer to popular opinion, I can't afford the EC lights demand)
* Small Ladder, Medium Ladder, Large Ladder (I defer to popular opinion)
* Medium Pegasus Ladder (I defer to popular opinion)
* Large Pegasus Ladder (I defer to popular opinion)

### Superior Models to Stock / Restock
Most of these are ones that Restock does differently, or does not yet touch. These should also be included.
* Rapier Engine
* Wheesley Turbofan
* Panther Turbofan
* Whiplash Ramjet
* Linear RCS
* RCS Thruster Block
* Shielded Docking Port
* Accelerometer Model
* M700 Scanner (not used in RP-1 currently)
* All Ladders

### Support as Part Variants Possibly
* Modular Girder Adapter
* Modular Girder Segment
* Modular Girder XL

**Comment by PhineasFreak** [comment]
Other notes as part of a personal "RO-Restock-VSR" integration foray:

* VSR could easily be updated to just add the extra parts. Just delete the "PathPatches" and "Squad" folders (although some Squad RO patches should be updated to reflect the removal of the VSR models but the Squad engine patches are going to be rewritten from scratch).
* RestockPlus has much better small RCS modules, better models for the CASTOR I to IV SRM family (maybe extendable to the ORION series), proper 4 segment RSRM (although we will be missing the "stock" 5 segment version).
* VSR has **much** better solar arrays. The Restock ones look..child-ish in comparison, with the exception of the ISS-alike SAW. RestockPlus could provide more solar arrays in the future, as the current selection is severely limited.
* The Restock versions of the LV-1 is much better than the VSR one (allows more realistic versions of the Generic 1 & 2 kN Thrusters, along with TD-339 and Aerobee).
* VSR has better models for the generic separation motors (can be used easier without clipping).
* With the exception of engine models not replicated by other mods (Astris & BE-4) most engines can easily be covered by ROEngines (or many other recommended/suggested mods). So, keeping around extra (and mostly inaccurate) engine models is not something that i would consider.
* The VSR VA command module is (was?) a popular solution when someone needed a 3-person lunar-rated reentry capsule but does not want to install the full FASA pack (and muck around trying to remove the rest of the parts) for the Apollo (ROCapsules might have that covered though). I'd say that it should be kept (or be integrated to ROCapsules).
* Restock provides the capability of completely removing part models & textures from the GameDatabase via blacklists. This could be used to remove parts from virtually any other installed mod, probably even from Restock itself (currently, parts are "deleted" by removing just the part config via MM).

(holy cow, a wall of text!)

**Comment by kurgut** [comment]
However, for restock & restock plus, I noticed on almost every engine some inconsistencies in relation to FAR:
eg, take the lvt-45, the botail variant should give way better performance (aera cross section smthg, don't remember exactly the name, it's in transonic design tab), whereas it doesn't. 
the " standard mounted" version does however : can see it when displaying debug voxels, which are consequently kinda messed up.
Don't know if it matters, but it did at leat when I tested it in stock.

**Comment by PhineasFreak** [comment]
What version of FAR are you using? The latest FAR release includes fixes for various voxelization issues, including many that were previously present on 100% stock parts but were caught just because of RS.

**Comment by kurgut** [comment]
0.15.10.1, from few days, as latest restock

**Comment by PhineasFreak** [comment]
That sounds like a bug. I'd report the findings to the Restock repository.

**Comment by blowfishpro** [comment]
All Restock parts use the original part’s collider at least for the default variant.  They may need to force FAR to use meshes over colliders.

**Comment by MikeOnTea** [comment]
>     * Communotron 88-88-V (remove, not currently used -> Pending Comms Overhaul?)

I've looked at all the antennas and i slightly prefer the restock versions, i don't think we need to keep vens here for anything. Restock+ adds a few more antennas including one that is similar but looks better than the 88-88 from Vens.

**Comment by PhineasFreak** [comment]
> They don't use Ven's LV-1 model but its OMS-Large model, which i think is very decent.

My bad, forgot that you updated the parts to use the "Astris" model. I concur, the new model is better.

> Also, shouldn't parts rather be added to RO (or some RO suggested mod created by us) instead of RP0 instead?

Of course! I have been pointing that out for a long time, especially since many newer RP-1 parts are locked behind it and making it impossible to open craft files if you don't have RP-1 (+ a ton of other dependent mods) installed.

**Comment by pap1723** [comment]
@MikeOnTea you are right about the engines and what I actually did was add the Vens OMS and Sepatron model to RO since they are so much better.

As to why parts are not typically added to RO... RO has no dependencies other than what is necessary for it to function. That isn't going to change. On the other hand, RP-1 requires Vens and SXT so having new parts added to it makes sense.

**Comment by PhineasFreak** [comment]
And again, RP-1 is about:

* Career progression
* Science balance
* Parts tech tree placement
* Parts pricing
* Contracts

And not for:

* Part physical sizing
* Part mechanical/thermal strength
* Part resource configuration

Otherwise why not just merge the two mods? It will be much easier, since you could configure directly a part for both the above major categories and cut down on repository maintenance. But a long time ago they got separated, and for good reasons (nothing in common, ease of mod maintenance, can be released even if the other mod does not require updating, cuts down on dependent mods that need to be installed if someone does not want career mode).

Now, what happens if you have RO parts in RP-1? If someone wants these parts then he/she is forced to install RP-1. And Contract Configurator. And CBK. And KCT. And having to apply 25000 MM patches instead of just 8000. And having unnecessary overhead on sandbox mode.

Just my 2 cents.

**Comment by DRVeyl** [comment]
Given VSR is actively maintained again, and we have resolved some compatibility issues with ReStock as well, is this still an open issue?  I've done a fair bit of work on the RO side addressing ReStock + VSR/VNP patching to ensure consistent part stats.  VSR core and ReStock do not create any parts, just improve the visual appearance.  I don't know that making either a requirement makes sense.

We have made VNP a requirement for some parts.  There are a few custom parts we make that we construct from multiple VSR/VNP models.  (Currently, some compressors on a few AJE engines.)

If you want to address preferences for one model supplier over another, especially where your preferences change based on the part... I think we'll suffer a lot of headaches trying to sort out the direct conflict between ReStock and the VSR/Squad patches.  Also that they (correctly) mark themselves as conflicting via CKAN.
